src/aaz_dev/cli/controller/portal_cli_generator.py

```diff
@@ -0,0 +1,359 @@
+        rp_name = resource.rp_name
+        swagger_path = resource.swagger_path
+        cmd_portal_info['rp_name'] = rp_name
+        swagger_provider_ind = swagger_path.rfind(rp_name)
+        provider_ind = 0
+        while swagger_provider_ind == -1 and provider_ind < len(self.PROVIDERS_LIST):
+            swagger_provider_ind = swagger_path.rfind(self.PROVIDERS_LIST[provider_ind])
+            provider_ind += 1
+        if swagger_provider_ind == -1:
+            print("please check cmd " + " ".join(leaf.names) + " resource id: " + swagger_path)
+            return
+        resource_paths = re.finditer(self.RESOURCE_PATH_PATTERN, swagger_path[swagger_provider_ind:])
+        folder_end_ind = swagger_path[swagger_provider_ind:].rfind("{", )
+        resource_type = ""
+        for resource_match in resource_paths:
+            resource_folder = resource_match.group(1)
+            if len(resource_type) == 0:
+                resource_type += resource_folder
+            elif resource_match.span(1)[0] < folder_end_ind:
+                resource_type += "/" + resource_folder
+        cmd_portal_info['resourceType'] = resource_type
+
+    def fill_api_version(self, cmd_portal_info, target_version):
+        if target_version.name is None:
+            raise exceptions.ResourceNotFind("version name not exist")
+        cmd_portal_info['apiVersion'] = target_version.name
+
+    def generate_base_learn_more(self, cmd_portal_info, leaf):
+        """
+        cmd: az xx yy zz
+        leaf.names: [xx, yy, zz]
+        help link: https://learn.microsoft.com/cli/azure/xx/yy
+        """
+        base_help_url = self.DOC_ROOT_NAME + "/" + "/".join(leaf.names[:-1])
+        cmd_portal_info['learnMore'] = {'url': base_help_url}
+
+    def generate_cmd_learn_more(self, help_info, leaf):
+        """
+        cmd name: az xx yy zz
+        leaf.names: [xx, yy, zz]
+        help link: https://learn.microsoft.com/cli/azure/xx/yy#az-xx-yy-zz
+        """
+        cmd_help_url = self.DOC_ROOT_NAME + "/" + "/".join(leaf.names[:-1]) + "#" + \
+                       "-".join([self.COMMAND_ROOT_NAME] + leaf.names)
+        help_info['learnMore'] = {'url' : cmd_help_url}
+
+    def fill_cmd_name(self, cmd_info, leaf):
+        cmd_name = " ".join([self.COMMAND_ROOT_NAME] + leaf.names)
+        cmd_info['name'] = cmd_name
+
+    def fill_cmd_desc(self, cmd_info, leaf):
+        cmd_info['description'] = leaf.help.to_primitive()['short']
+
+    def fill_cmd_path(self, cmd_info, cmd_cfg):
+        if cmd_cfg.resources is None:
+            raise exceptions.ResourceNotFind("Resource not exist")
+        resource = cmd_cfg.resources[-1]
+        cmd_info['path'] = resource.swagger_path
+
+    def fill_cmd_confirmation(self, cmd_info, cmd_cfg):
+        if cmd_cfg.confirmation is None:
+            cmd_info['confirmation'] = False
+        else:
+            cmd_info['confirmation'] = True
```

Comment:
I think the empty string should be false.
```suggestion
        if cmd_cfg.confirmation:
            cmd_info['confirmation'] = True
        else:
            cmd_info['confirmation'] = False
```